### PR TITLE
Add `creation_instant` as second sort key in tx pool.

### PR DESCRIPTION
**Pull request by AurelienFT**
https://github.com/FuelLabs/fuel-core/issues/1739

Add creation instant as second sort key in tx pool.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [x] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [x] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/services/txpool/src/txpool/tests.rs [hunk 1]
```diff
@@ -955,6 +955,78 @@ async fn sorted_out_tx_profitable_ratios() {
...
+    let txs = txpool.sorted_includable().collect::<Vec<_>>();
+
+    // This order doesn't match the lexicographical order of the tx ids
+    // and so it verifies that the txs are sorted by creation instant
+    // The newest tx should be first
+    assert_eq!(txs.len(), 4, "Should have 3 txs");
+    assert_eq!(txs[0].id(), tx4_id, "First should be tx4");
+    assert_eq!(txs[1].id(), tx3_id, "Second should be tx3");
+    assert_eq!(txs[2].id(), tx2_id, "Third should be tx2");
+    assert_eq!(txs[3].id(), tx1_id, "Fourth should be tx1");
```
Comment (Voxelot) [inline crates/services/txpool/src/txpool/tests.rs:1027]: If tx1 was inserted first, shouldn't it come before the later transactions?

We should prefer FIFO ordering assuming the tip doesn't jump the queue.

Comment (AurelienFT) [reply crates/services/txpool/src/txpool/tests.rs:1027]: Oh Okey I was thinking FIFO with out meaning getting pruned from the pool.

Comment (AurelienFT) [reply crates/services/txpool/src/txpool/tests.rs]: Fixed.

Comment (Voxelot) [reply crates/services/txpool/src/txpool/tests.rs]: thank you 🙌🏻

**Review by Voxelot — APPROVED** [review]

**Pull request merged**